ldms/src/ldmsd/ldmsd_updtr.c

```diff
@@ -537,28 +548,35 @@ void __ldmsd_prdset_lookup_cb(ldms_t xprt, enum ldms_lookup_status status,
 		assert(NULL == set);
 		status = (status < 0 ? -status : status);
 		if (status == ENOMEM) {
-			ldmsd_log(LDMSD_LERROR,
-				"Error %d in lookup callback for set '%s' "
-				"Consider changing the -m parameter on the "
-				"command line to a larger value. "
-				"The current value is %s\n",
-				status, prd_set->inst_name,
-				ldmsd_get_max_mem_sz_str());
+			if (! (prd_set->warned & LDMSD_PRDCR_SET_WARNED_NOMEM)) {
+				prd_set->warned |= LDMSD_PRDCR_SET_WARNED_NOMEM;
+				ldmsd_log(LDMSD_LERROR,
+					"Error %d in lookup callback for set '%s' "
+					"Consider changing the -m parameter on the "
+					"command line to a larger value. "
+					"The current value is %s\n",
+					status, prd_set->inst_name,
+					ldmsd_get_max_mem_sz_str());
+			}
 		} else if (status == EEXIST) {
-			ldmsd_log(LDMSD_LERROR, "Prdcr '%s': lookup failed asynchronously. "
-					"The set '%s' already exists. "
-					"It is likely that there are more "
-					"than one producers pointing to "
-					"the set.\n",
-					prd_set->prdcr->obj.name, prd_set->inst_name);
+			warn_once_prdcr_set_exists(prd_set);
 		} else {
-			ldmsd_log(LDMSD_LERROR,
-				  "Error %d in lookup callback for set '%s'\n",
-					  status, prd_set->inst_name);
+			if (! (prd_set->warned & LDMSD_PRDCR_SET_WARNED_LOOKUP)) {
```

Comment:
This is a configuration error, again it will make it seem like everything is ok when in fact, there is a serious configuration problem.

Reply:
Again, not hiding anything-- just logging it exactly once instead of every time the updater or reconnect loop re-executes.